src/data/equiv/fin.lean

```diff
@@ -37,13 +37,37 @@ def fin_two_equiv : fin 2 ≃ bool :=
   end,
   assume b, match b with tt := rfl | ff := rfl end⟩
 
-/-- Equivalence between `fin n.succ` and `option (fin n)` -/
+/-- Equivalence between `fin n.succ` and `option (fin n)`.
+This is a version of `fin.pred` that produces `option (fin n)` instead of
+requiring a proof that the input is not `0`. -/
 def fin_succ_equiv (n : ℕ) : fin n.succ ≃ option (fin n) :=
 ⟨λ x, fin.cases none some x, λ x, option.rec_on x 0 fin.succ,
   λ x, fin.cases rfl (λ i, show (option.rec_on (fin.cases none some (fin.succ i) : option (fin n))
     0 fin.succ : fin n.succ) = _, by rw fin.cases_succ) x,
   by rintro ⟨none | x⟩; [refl, exact fin.cases_succ _]⟩
 
+/-- An equivalence that removes `i` and maps it to `none`.
+This is a version of `fin.pred_above` that produces `option (fin n)` instead of
+requiring a proof that the input is not `i`. -/
+def fin_succ_above_equiv {n : ℕ} (i : fin n.succ) :
```

Comment:
This name is a bit confusing. Maybe something like `fin_succ_equiv_skipping`?

Reply:
It's named this for consistency with `fin_succ_equiv` above - `fin_***_equiv.symm` is a natural extension of `fin_***` in both cases.

Reply:
This doesn't actually make sense though because `****` changes.

Reply:
What do you mean by that? Note the `symm` in my explanation.

Reply:
I understand what you're going for, but I don't like it. I think you're misparsing the name `fin_succ_equiv`. It says `fin (n+1)` is equiv to something. The new definition also says `fin (n+1)` is equiv to something. It doesn't make sense to say "well `succ` is kind of like `pred` so let's just slip in `above` somewhere".

Reply:
Just `fin_succ_equiv'` might be a fine name too.

Reply:
I wasn't saying "succ is kind of like pred" (updated my comment above as I see you updated yours). I see your point though, the `fin_succ` in `fin_succ_equiv.symm` is ambiguous between `fin.succ` and `fin (succ n)`

Reply:
I feel like having the word `above` in here somewhere is important, because both directions of the equivalence use a definition with `above`, and that makes it easier to find this.

Reply:
I think that if people are search for this equiv, then they'll search for something involving `fin`, `equiv` and maybe `option` or `succ`. So even with tab-complete something like `fin_succ_equiv'` should be reasonably easy to find.
I think you can mention `succ_above` in the docstring.
But in general I agree with Reid's comments that this name is not optimal...

Reply:
I'm not so convinced that the name needs to be changed: we have a lot of bundled homs named `_hom`, so a bundled equiv named `_equiv` doesn't sound very weird to me.

Reply:
I ended up changing this anyway, I don't really care enough about the name, and with @jcommelin's comment it was two arguments against it.